runelite-client/src/main/java/net/runelite/client/ui/overlay/components/ProgressBarComponent.java

```diff
@@ -126,4 +139,27 @@ public Dimension render(Graphics2D graphics)
 		bounds.setSize(dimension);
 		return dimension;
 	}
+
+	public void setPaused(boolean pause)
```

Comment:
This component is meant to be generic, not including logic for pausing. This should either be handled in the plugin itself (saving state and updating the ProgressBarComponent it is using over time) or by creating a new class which extends this with the desired pausing functionality.